core/src/main/java/org/fao/geonet/kernel/datamanager/base/BaseMetadataManager.java

```diff
@@ -1329,4 +1310,18 @@ public boolean isValid(Integer id) {
         }
         return true;
     }
+
+    private boolean hasResult(SearchResponse searchResponse) {
+        return searchResponse.getHits().getTotalHits().value > 0;
+    }
+    SearchResponse searcherForReferencingMetadata(ServiceContext context, AbstractMetadata metadata) throws Exception {
+        StringBuilder query = new StringBuilder(String.format("xlink:*%s*", metadata.getUuid()));
+        query.append(String.format(" AND (%s:y OR %s:n)", IS_TEMPLATE, IS_TEMPLATE).toString());
+        return this.searchManager.query(query.toString(), null, 0, 0);
+    }
+
+    boolean hasReferencingMetadata(ServiceContext context, AbstractMetadata metadata) throws Exception {
+        return hasResult(searcherForReferencingMetadata(context, metadata));
```

Comment:
maybe should searcherForReferencingMetadata (by now this is more a searchResponse than a searcher) and hasResult be inlined here.